Dockerfile

```diff
@@ -1,12 +1,27 @@
-ARG SOURCE_BINARY=bin/manager
+FROM golang:1.17 as builder
+
+ARG TAG
+ARG COMMIT
+ARG REPO_INFO
+WORKDIR /workspace
+
+ADD go.mod .
+ADD go.sum .
+RUN go mod download
+
+COPY pkg/ pkg/
+COPY api/ api/
+COPY cmd/ cmd/
+COPY controllers/ controllers/
+COPY main.go main.go
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+FROM gcr.io/distroless/static:nonroot AS distroless
 WORKDIR /
-COPY ${SOURCE_BINARY} .
-EXPOSE 8080 8443
-USER nonroot:nonroot
+COPY --from=builder /workspace/manager .
+USER 65532:65532
```

Comment:
Why is the uid `65532`? I see the `nonroot` user uid should be `1002` and probably can still be referenced simply as `USER nonroot:nonroot`

Reply:
df09a02181b00d8c102fe6001982b2265490f072
no permission issue so far.